setup.py

```diff
@@ -66,6 +66,7 @@ def read_requirements(req="base.txt"):
         "NOTICE",
         "THIRD_PARTY_LICENSES",
     ),
+    python_requires=">=3.7, <=4.0, !=4.0",
```

Comment:
Should we restrict this to only supported versions instead of auto enabling new python versions under 4.0?

Reply:
Rethinking this. Since this repo is a library, leaving it open like this is likely fine. We may run into problems if we use newer lang features but we have to support the min version anyways.

Reply:
Why `<=4.0` and `!=4.0` instead of just `<4.0`?

(Or better yet only `>=3.7` https://iscinumpy.dev/post/bound-version-constraints/)